### ISSUE TITLE
Request for announcement of weekday along with the date in screen reader(voice over)

### ISSUE THREAD

**Issue by SahilSetia2512** [wojtekmaj/react-calendar#961]
### Before you start - checklist

- [X] I have checked if this feature request is not already reported

### Description

Request for announcement of weekday along with the date in screen reader(voice-over). Currently, it announces the date only.

### Proposed solution

I checked through the code and saw that while giving the aria-label in <tile> component we are evaluating the label that should be announced. For that formatLongDate function, we are not specifying weekDay as long in formatting options. We need to add support for that.

<img width="1691" alt="Screenshot 2024-12-02 at 4 37 46 PM" src="https://github.com/user-attachments/assets/5f003e2a-2463-438a-88bf-ab482e72ed90">

https://github.com/user-attachments/assets/522fae2e-d600-4f3f-93e0-4da1147b2d2f

### Additional information

**Without weekday option(Current behaviour)**

```
const date = newDate('2023-12-02');
const formatter = newIntl.DateTimeFormat('en-US', formatLongDateOptions);
console.log(formatter.format(date)); // Output: "December 2, 2023" (in the U.S. English locale)
```
 
**With weekday option(Proposed behaviour)**

```
const options = { 
  weekday: 'long', 
  day: 'numeric', 
  month: 'long', 
  year: 'numeric' 
};
console.log(date.toLocaleDateString('en-US', options)); // Output: "Saturday, December 2, 2023"
```
